tests/components/modbus/test_light.py

```diff
@@ -307,21 +347,336 @@ async def test_light_service_turn(
     [
+        (
+            {
+                CONF_LIGHTS: [
+                    {
+                        CONF_NAME: TEST_ENTITY_NAME,
+                        CONF_ADDRESS: 1234,
+                        CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                        CONF_SCAN_INTERVAL: 0,
+                    }
+                ]
+            },
+            None,
+            None,
+            None,
+            None,
+        ),
+        (
+            {
+                CONF_LIGHTS: [
+                    {
+                        CONF_NAME: TEST_ENTITY_NAME,
+                        CONF_ADDRESS: 1234,
+                        CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                        CONF_SCAN_INTERVAL: 0,
+                    }
+                ]
+            },
+            155,
+            3000,
+            None,
+            None,
+        ),
+        (
+            {
+                CONF_LIGHTS: [
+                    {
+                        CONF_NAME: TEST_ENTITY_NAME,
+                        CONF_ADDRESS: 1234,
+                        CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                        CONF_SCAN_INTERVAL: 0,
+                        CONF_COLOR_TEMP_REGISTER: 2,
+                    }
+                ]
+            },
+            128,
+            3000,
+            None,
+            20,
+        ),
+        (
+            {
+                CONF_LIGHTS: [
+                    {
+                        CONF_NAME: TEST_ENTITY_NAME,
+                        CONF_ADDRESS: 1234,
+                        CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                        CONF_SCAN_INTERVAL: 0,
+                        CONF_BRIGHTNESS_REGISTER: 1,
+                        CONF_COLOR_TEMP_REGISTER: 2,
+                    }
+                ]
+            },
+            128,
+            2000,
+            50,
+            0,
+        ),
+        (
+            {
+                CONF_LIGHTS: [
+                    {
+                        CONF_NAME: TEST_ENTITY_NAME,
+                        CONF_ADDRESS: 1234,
+                        CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                        CONF_SCAN_INTERVAL: 0,
+                        CONF_BRIGHTNESS_REGISTER: 1,
+                    }
+                ]
+            },
+            128,
+            None,
+            50,
+            None,
+        ),
+    ],
+)
+async def test_color_temp_brightness_light(
+    hass,
+    mock_modbus_ha,
+    brightness_input,
+    color_temp_input,
+    expected_brightness,
+    expected_color_temp,
+):
+    """Test Modbus Light color temperature and brightness."""
+    assert hass.states.get(ENTITY_ID).state == STATE_OFF
+
+    service_data = {ATTR_ENTITY_ID: ENTITY_ID}
+
+    if brightness_input is not None:
+        service_data[ATTR_BRIGHTNESS] = brightness_input
+
+    if color_temp_input is not None:
+        service_data[ATTR_COLOR_TEMP_KELVIN] = color_temp_input
```

Comment:
since we should not have conditions in tests (some old tests still have them, but newer ones shouldn't), lets put the whole `service_data` in the test params.
You can just set it as a object in the test params like 
```python
{ATTR_BRIGHTNESS: 128}
```
and
```python
{ATTR_BRIGHTNESS: 128, ATTR_COLOR_TEMP_KELVIN: 3000}
```

Then in the test body you can merge the param with the entity_id part:
```python
        {ATTR_ENTITY_ID: ENTITY_ID, **service_data},
```